lib/extensions/ar_visible_attribute.rb

```diff
@@ -0,0 +1,23 @@
+module ArVisibleAttribute
+  extend ActiveSupport::Concern
+
+  included do
+    cattr_accessor :hidden_attribute_names
+    self.hidden_attribute_names = []
+  end
+
+  class_methods do
+    # @param [String|Symbol] attribute name of attribute to be hidden from the api and reporting
+    # this attribute is accessible to all ruby methods. But it is not advertised.
+    # we do this when deprecating an attribute or when introducing an internal attribute
+    def hide_attribute(attribute)
+      hidden_attribute_names << attribute.to_s
```

Comment:
I think this line needs to be something like `self.hidden_attribute_names += [attribute.to_s]`

Reply:
and for anyone curious about why the `self` is necessary, the first comment on https://apidock.com/rails/Class/class_attribute addresses some of that question.